File: leetcode/AIcoding/3.DeepSeek-v3/python_solutions/3394-check-if-grid-can-be-cut-into-sections.py
class Solution:
    def possibleToStamp(self, grid: List[List[int]], stampHeight: int, stampWidth: int) -> bool:
        m, n = len(grid), len(grid[0])
        prefix = [[0] * (n + 1) for _ in range(m + 1)]
        diff = [[0] * (n + 1) for _ in range(m + 1)]

        for i in range(m):
            for j in range(n):
                prefix[i + 1][j + 1] = prefix[i][j + 1] + prefix[i + 1][j] - prefix[i][j] + grid[i][j]

        for i in range(m - stampHeight + 1):
            for j in range(n - stampWidth + 1):
                x1, y1 = i, j
                x2, y2 = i + stampHeight - 1, j + stampWidth - 1
                total = prefix[x2 + 1][y2 + 1] - prefix[x1][y2 + 1] - prefix[x2 + 1][y1] + prefix[x1][y1]
                if total == 0:
                    diff[x1][y1] += 1
                    diff[x1][y2 + 1] -= 1
                    diff[x2 + 1][y1] -= 1
                    diff[x2 + 1][y2 + 1] += 1

        for i in range(m):
            for j in range(n):
                if i > 0:
                    diff[i][j] += diff[i - 1][j]
                if j > 0:
                    diff[i][j] += diff[i][j - 1]
                if i > 0 and j > 0:
                    diff[i][j] -= diff[i - 1][j - 1]
                if grid[i][j] == 0 and diff[i][j] == 0:
                    return False
        return True

File: leetcode/AIcoding/3.DeepSeek-v3/javascript_solutions/3394-check-if-grid-can-be-cut-into-sections.js
var possibleToStamp = function(grid, stampHeight, stampWidth) {
    const m = grid.length;
    const n = grid[0].length;

    const prefix = Array(m + 1).fill().map(() => Array(n + 1).fill(0));
    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n; j++) {
            prefix[i + 1][j + 1] = grid[i][j] + prefix[i][j + 1] + prefix[i + 1][j] - prefix[i][j];
        }
    }

    const diff = Array(m + 2).fill().map(() => Array(n + 2).fill(0));
    for (let i = stampHeight; i <= m; i++) {
        for (let j = stampWidth; j <= n; j++) {
            const x = i - stampHeight + 1;
            const y = j - stampWidth + 1;
            const sum = prefix[i][j] - prefix[x - 1][j] - prefix[i][y - 1] + prefix[x - 1][y - 1];
            if (sum === 0) {
                diff[x][y]++;
                diff[x][j + 1]--;
                diff[i + 1][y]--;
                diff[i + 1][j + 1]++;
            }
        }
    }

    for (let i = 1; i <= m; i++) {
        for (let j = 1; j <= n; j++) {
            diff[i][j] += diff[i - 1][j] + diff[i][j - 1] - diff[i - 1][j - 1];
            if (grid[i - 1][j - 1] === 0 && diff[i][j] === 0) {
                return false;
            }
        }
    }

    return true;
};